File: app/src/main/java/br/com/udacity/ruyano/recipes/networking/repositories/RecipeRepository.java
package br.com.udacity.ruyano.recipes.networking.repositories;

import java.util.ArrayList;
import java.util.List;

import androidx.lifecycle.MutableLiveData;
import br.com.udacity.ruyano.recipes.models.Recipe;
import br.com.udacity.ruyano.recipes.networking.RetrofitConfig;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class RecipeRepository {

    private final MutableLiveData<List<Recipe>> recipesLiveData;

    public RecipeRepository() {
        this.recipesLiveData = new MutableLiveData<>();

    }

    public void getRecipes() {
        RetrofitConfig.getInstance().getApi().getRecipes().enqueue(new Callback<List<Recipe>>() {
            @Override
            public void onResponse(Call<List<Recipe>> call, Response<List<Recipe>> response) {
                recipesLiveData.postValue(response.body());

            }

            @Override
            public void onFailure(Call<List<Recipe>> call, Throwable t) {
                recipesLiveData.postValue(new ArrayList<Recipe>());

            }
        });

    }

    public MutableLiveData<List<Recipe>> getRecipesLiveData() {
        return recipesLiveData;

    }
}

File: app/src/main/java/br/com/udacity/ruyano/recipes/networking/repositories/RecipeRepository.kt
package br.com.udacity.ruyano.recipes.networking.repositories

import androidx.lifecycle.MutableLiveData
import br.com.udacity.ruyano.recipes.models.Recipe
import br.com.udacity.ruyano.recipes.networking.RetrofitConfig
import br.com.udacity.ruyano.recipes.networking.services.IAPIService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class RecipeRepository(
        private val api: IAPIService
) {
    val recipesLiveData: MutableLiveData<List<Recipe>> = MutableLiveData()

    fun recipes() {
        api.recipes.enqueue(object : Callback<List<Recipe>> {
            override fun onResponse(call: Call<List<Recipe>>, response: Response<List<Recipe>>) {
                recipesLiveData.postValue(response.body())
            }

            override fun onFailure(call: Call<List<Recipe>>, t: Throwable) {
                recipesLiveData.postValue(ArrayList())
            }
        })
    }
}